File: src/services/customer/profileService.js
// services/customer/profile.js
import api from '../api';

//  Get Customer Profile
export const getCustomerProfile = async token => {
  try {
    const response = await api.get('/nukkad/api/customer/v1/get/profile', {
      headers: {
        Authorization: `Bearer ${token}`,
      },
    });

    console.log(' Get Profile API Success:', {
      status: response.status,
      data: response.data,
    });

    const user = response.data || {};

    // Parse name -> firstName, lastName
    const [firstName = '', ...rest] = user.name?.split(' ') || [];
    const lastName = rest.join(' ');
    return {
      ...user,
      firstName,
      lastName,
      DOB: user.dob,
    };
  } catch (error) {
    console.error(' Get Profile API Error:', {
      message: error.message,
      status: error.response?.status,
      data: error.response?.data,
    });

    throw error;
  }
};

//  Update Customer Profile
export const updateCustomerProfile = async (payload, token) => {
  try {
    // Only send JSON now; image URL is included in the JSON
    const requestBody = {
      name: payload.name || '',
      email: payload.email || '',
      dob: payload.dob || '',
      mobileNumber: payload.mobileNumber || '',
      profileImageUrl: payload.image || '', // Firebase URL
    };
    console.log('request body from updateApi', requestBody);

    const response = await api.put(
      '/nukkad/api/customer/v1/update',
      requestBody,
      {
        headers: {
          'Content-Type': 'application/json',
          Authorization: `Bearer ${token}`,
        },
      },
    );

    console.log('✅ Update success:', response.data);
    return response.data?.data;
  } catch (error) {
    console.error('❌ Update error:', {
      message: error.message,
      status: error.response?.status,
      data: error.response?.data,
      request: payload,
    });
    throw new Error(
      error.response?.data?.message || 'Failed to update customer profile',
    );
  }
};

File: lib/services/profile_service.dart
import 'dart:convert';
import 'package:flutter_dotenv/flutter_dotenv.dart';
import 'package:http/http.dart' as http;
import '../common/logger.dart';
import '../db/local_storage.dart';
import '../models/profile_model.dart';

class ProfileService {


  static Future<ProfileModel?> getProfile() async {

    final token = LocalStorageService.getToken();
    final baseUrl = dotenv.env['BASE_URL'];
    if (token == null) {
      log.e("Token missing");
      return null;
    }
    log.d("PROFILE URL: $baseUrl/customer/v1/get/profile");
    log.d("PROFILE TOKEN: $token");
    final response = await http.get(
      Uri.parse('$baseUrl/customer/v1/get/profile'),
      headers: {
        "Authorization": "Bearer $token",
        "Content-Type": "application/json"
      },
    );

    log.d("Profile Status: ${response.statusCode}");
    log.d("Profile Body: ${response.body}");

    if (response.statusCode == 200) {
      final data = jsonDecode(response.body);

      return ProfileModel.fromJson(data);
    }

    return null;
  }
  static Future<bool> updateProfile({
    required String name,
    required String email,
    required String dob,
    String? profileImageUrl,
  }) async {

    final token =  LocalStorageService.getToken();
    final baseUrl = dotenv.env['BASE_URL'];
    final response = await http.put(
      Uri.parse("$baseUrl/customer/v1/update"),
      headers: {
        "Authorization": "Bearer $token",
        "Content-Type": "application/json",
      },
      body: jsonEncode({
        "name": name,
        "email": email,
        "dob": dob,
        "profileImageUrl": profileImageUrl
      }),
    );

    log.d("Update Profile Status: ${response.statusCode}");
    log.d("Update Profile Body: ${response.body}");

    return response.statusCode == 200;
  }
}